include/boost/gil/image_processing/numeric.hpp

```diff
@@ -295,6 +295,120 @@ inline void compute_hessian_entries(
+                nabla[(std::size_t)direction::west](x, y)[channel_index] = view(x - 1, y)[channel_index] - view(x, y)[channel_index];
+                nabla[(std::size_t)direction::east](x, y)[channel_index] = view(x + 1, y)[channel_index] - view(x, y)[channel_index];
+
+                nabla[(std::size_t)direction::north_east](x, y)[channel_index] = view(x + 1, y - 1)[channel_index] - view(x, y)[channel_index];
+                nabla[(std::size_t)direction::south_east](x, y)[channel_index] = view(x + 1, y + 1)[channel_index] - view(x, y)[channel_index];
+                nabla[(std::size_t)direction::south_west](x, y)[channel_index] = view(x - 1, y + 1)[channel_index] - view(x, y)[channel_index];
+                nabla[(std::size_t)direction::north_west](x, y)[channel_index] = view(x - 1, y - 1)[channel_index] - view(x, y)[channel_index];
+            }
+        }
+    }
+}
+
+template <typename View>
+void calculate_diffusvity(std::vector<View> nablas, double kappa, const std::vector<View> diffusivities)
+{
+    using pixel_type = typename View::value_type;
+    using channel_type = typename channel_type<View>::type;
+    BOOST_ASSERT(nablas.size() == diffusivities.size());
+    for (std::size_t i = 0; i < nablas.size(); ++i) {
+        gil::transform_pixels(nablas[i], diffusivities[i], [kappa](pixel_type p)
+        {
+            auto op = [kappa](channel_type value)
+            {
+                value /= static_cast<channel_type>(kappa);
+                auto result = std::exp(-value * value);
+                return result;               
+            };
+            pixel_type result_pixel;
+            static_transform(p, result_pixel, op);
+            return result_pixel;
+        });
+    }
+}
+} // namespace boost::gil::detail
+
+/// \brief Performs diffusion according to Perona-Malik equation
+/// 
+/// Anisotropic diffusion is a smoothing algorithm that respects
+/// edge boundaries and can work as an edge detector if suitable
+/// iteration count is set and grayscale image view is used
+/// as an input
+template <typename InputView, typename OutputView>
+void anisotropic_diffusion(InputView input, unsigned int num_iter, double delta_t, double kappa, OutputView output)
+{
+    gil::copy_pixels(input, output);
+    using element_type = typename OutputView::value_type;
+    using computation_image_type = image<element_type>;
+    using channel_type = typename channel_type<OutputView>::type;
+
+    for (unsigned int i = 0; i < num_iter; ++i) {
+        std::vector<computation_image_type> nabla_images(8, computation_image_type(input.dimensions()));
+        std::vector<typename computation_image_type::view_t> nabla;
+        std::transform(nabla_images.begin(), nabla_images.end(),
+                       std::back_inserter(nabla), [](computation_image_type& img)
+                        {
+                            return gil::view(img);
+                        });
```

Comment:
There is a bit too much blank space to the left. I'd either re-indent towards left 3 times or just do

```cpp
 std::transform(nabla_images.begin(), nabla_images.end(),
    std::back_inserter(nabla), [](computation_image_type& img) { return gil::view(img); });
```